apps/app_voicemail.c

```diff
@@ -3919,30 +4095,80 @@ static SQLHSTMT generic_prepare(struct odbc_obj *obj, void *data)
 static void odbc_update_msg_id(char *dir, int msg_num, char *msg_id)
 {
 	SQLHSTMT stmt;
-	char sql[PATH_MAX];
+	char *sql = MAKE_SQL_PTRA("UPDATE %s SET msg_id=? WHERE dir=? AND msgnum=?");
 	struct odbc_obj *obj;
 	char msg_num_str[20];
 	char *argv[] = { msg_id, dir, msg_num_str };
 	struct generic_prepare_struct gps = { .sql = sql, .argc = 3, .argv = argv };
+	SCOPE_ENTER(3, "dir: %s msg_num: %d msg_id: %s\n", dir, msg_num, msg_id);
 
 	obj = ast_odbc_request_obj(odbc_database, 0);
 	if (!obj) {
-		ast_log(LOG_WARNING, "Unable to update message ID for message %d in %s\n", msg_num, dir);
-		return;
+		SCOPE_EXIT_LOG_RTN(LOG_WARNING, "Unable to update message ID for message %d in %s\n", msg_num, dir);
 	}
 
 	snprintf(msg_num_str, sizeof(msg_num_str), "%d", msg_num);
-	snprintf(sql, sizeof(sql), "UPDATE %s SET msg_id=? WHERE dir=? AND msgnum=?", odbc_table);
 	stmt = ast_odbc_prepare_and_execute(obj, generic_prepare, &gps);
 	if (!stmt) {
 		ast_log(LOG_WARNING, "SQL Execute error!\n[%s]\n\n", sql);
 	} else {
 		SQLFreeHandle(SQL_HANDLE_STMT, stmt);
 	}
 	ast_odbc_release_obj(obj);
-	return;
+	SCOPE_EXIT_RTN();
 }
 
+#define AUDIO_ON_DISK_MAGIC "AUDMAGIC"
+#define AUDIO_ON_DISK_MAGIC_LEN 8
```

Comment:
#define AUDIO_ON_DISK_MAGIG_LEN   (sizeof(AUDIO_ON_DISK_MAGIC) - 1) ??

Reply:
We already know the length and it doesn't make sense to run sizeof every time the macro is referenced.

Reply:
The sizeof would be done at compile by the preprocessor

Reply:
same applies for strlen() nowadays if optimization is enabled, if the string is a compile-time constant strlen() reduces to a compile-time constant ...